docs/samples/Microsoft.ML.Samples/Dynamic/SsaChangePointDetectorTransform.cs

```diff
@@ -66,23 +75,127 @@ public static void SsaChangePointDetectorTransform()
+            var engine = model.CreateTimeSeriesPredictionFunction<SsaChangePointData, ChangePointPrediction>(ml);
+
+            // Start streaming new data points with no change point to the prediction engine.
+            Console.WriteLine($"Output from ChangePoint predictions on new data:");
+            Console.WriteLine("Data\tAlert\tScore\tP-Value\tMartingale value");
+            ChangePointPrediction prediction = null;
+            for (int i = 0; i < 5; i++)
+            {
+                var value = i;
+                prediction = engine.Predict(new SsaChangePointData(value));
+                Console.WriteLine("{0}\t{1}\t{2:0.00}\t{3:0.00}\t{4:0.00}", value, prediction.Prediction[0], prediction.Prediction[1], prediction.Prediction[2], prediction.Prediction[3]);
+            }
+
+            // Now stream data points that reflect a change in trend.
+            for (int i = 0; i < 5; i++)
+            {
+                var value = (i + 1) * 100;
+                prediction = engine.Predict(new SsaChangePointData(value));
+                Console.WriteLine("{0}\t{1}\t{2:0.00}\t{3:0.00}\t{4:0.00}", value, prediction.Prediction[0], prediction.Prediction[1], prediction.Prediction[2], prediction.Prediction[3]);
+            }
+
+            // Now we demonstrate saving and loading the model.
+
+            // Save the model that exists within the prediction engine.
+            // The engine has been updating this model with every new data point.
+            var modelPath = "model.zip";
+            engine.CheckPoint(ml, modelPath);
+
+            // Load the model.
+            using (var file = File.OpenRead(modelPath))
+                model = TransformerChain.LoadFrom(ml, file);
+
+            // We must create a new prediction engine from the persisted model.
+            engine = model.CreateTimeSeriesPredictionFunction<SsaChangePointData, ChangePointPrediction>(ml);
+
+            // Run predictions on the loaded model.
+            for (int i = 0; i < 5; i++)
+            {
+                var value = (i + 1) * 100;
+                prediction = engine.Predict(new SsaChangePointData(value));
+                Console.WriteLine("{0}\t{1}\t{2:0.00}\t{3:0.00}\t{4:0.00}", value, prediction.Prediction[0], prediction.Prediction[1], prediction.Prediction[2], prediction.Prediction[3]);
+            }
+
+            // Output from ChangePoint predictions on new data:
+            // Data    Alert   Score   P-Value Martingale value
+            // 0       0     - 1.01    0.50    0.00
+            // 1       0     - 0.24    0.22    0.00
+            // 2       0     - 0.31    0.30    0.00
+            // 3       0       0.44    0.01    0.00
+            // 4       0       2.16    0.00    0.24
+            // 100     0      86.23    0.00    2076098.24
+            // 200     0     171.38    0.00    809668524.21
+            // 300     1     256.83    0.01    22130423541.93    <-- alert is on, note that delay is expected
```

Comment:
>93    <-- alert is on, note that delay is expected [](start = 59, length = 50)

Can you please add the indicator when the mode was saved/loaded?

Reply:
Good idea, I'll add that.

---
In reply to: [238001208](https://github.com/dotnet/machinelearning/pull/1788#discussion_r238001208) [](ancestors = 238001208)